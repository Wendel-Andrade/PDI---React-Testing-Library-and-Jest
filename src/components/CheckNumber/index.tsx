import { ChangeEvent, useState } from "react";

export function CheckNumber() {
  const [number, setNumber] = useState<string>("0");
  const [errorMessage, setErrorMessage] = useState<string>("");

  const isEven = parseInt(number) % 2 === 0;

  function checkNumber(event: ChangeEvent<HTMLInputElement>) {
    const rawValue = event.target.value;
    const numberValue = parseInt(rawValue);

    setNumber(rawValue);

    if (isNaN(numberValue)) {
      setErrorMessage("Please enter another number");
      return;
    }

    setErrorMessage("");
  }

  return (
    <div>
      <h1>Component even or odd</h1>
      <input
        type="text"
        name="number"
        placeholder="enter the number"
        value={number}
        onChange={(event) => checkNumber(event)}
      />
      <p role="presentation">{isEven ? "Even" : "Odd"}</p>
      <p role="alert">{errorMessage}</p>
    </div>
  );
}
