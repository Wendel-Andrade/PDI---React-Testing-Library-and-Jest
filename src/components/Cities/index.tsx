import { useState } from "react";

export function Cities() {
  const [cities, setCities] = useState<string[]>([]);
  const [cityToCheck, setCityToCheck] = useState<string>("");
  const [isCityPresent, setIsCityPresent] = useState<boolean>(false);

  function initalizeCityDataBase() {
    setCities(["Vienna", "San Juan", "New York", "Tokyo"]);
  }

  function clearCityDataBase() {
    setCities([]);
  }

  function isCity(inputCity: string) {
    const lowerCaseCity = inputCity.toLowerCase();

    return cities.some((city) => city.toLowerCase() === lowerCaseCity);
  }

  function handleCityCheck() {
    setIsCityPresent(isCity(cityToCheck));
  }

  return (
    <div>
      <h3>Cities</h3>
      <button onClick={initalizeCityDataBase}>Create Cities</button>
      <button onClick={clearCityDataBase}>Clear cities</button>

      <ul>
        {cities?.map((city, index) => (
          <li key={index} style={{ listStyle: "none" }}>
            <strong>CITY:</strong> <span>{city}</span>
          </li>
        ))}
      </ul>
      <div>
        <input
          type="text"
          value={cityToCheck}
          onChange={(e) => setCityToCheck(e.target.value)}
          placeholder="Enter city name"
        />
        <button onClick={handleCityCheck}>Check City</button>
        {isCityPresent ? <p>City is present!</p> : <p>City is not present.</p>}
      </div>
    </div>
  );
}
