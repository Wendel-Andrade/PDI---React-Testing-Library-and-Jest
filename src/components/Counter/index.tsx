import { useCounter } from "../../hooks/useCounter";

export function Counter() {
  const { counter, increment, decrement, resetCounter } = useCounter();

  return (
    <>
      <button data-testid="increment-button" onClick={increment}>
        Increment
      </button>
      <button data-testid="decrement-button" onClick={decrement}>
        Decrement
      </button>
      <button data-testid="reset-counter-button" onClick={resetCounter}>
        Reset Counter
      </button>
      <p data-testid="counter-display">Count: {counter}</p>
    </>
  );
}
