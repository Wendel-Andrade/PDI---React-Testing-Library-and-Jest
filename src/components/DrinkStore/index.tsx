import { useEffect, useState } from "react";

export const drinks: Record<string, string> = {
  lemon: "lemon",
  strawberry: "strawberry",
  peach: "peach",
};

export function deliverDrink(): Promise<string> {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      resolve("Drink Delivered!");
    }, 1000);

    setTimeout(() => {
      reject(new Error("Drink Not Delivered!"));
    }, 1000);
  });
}

export function DrinkStore() {
  const [drink, setDrink] = useState<string>("");

  useEffect(() => {
    deliverDrink().then((result) => {
      return setDrink(result);
    });
  }, []);

  return (
    <div>
      <h1>Examples</h1>
      <p>To be example: {drinks.lemon}</p>
      <p>{drink}</p>
    </div>
  );
}
