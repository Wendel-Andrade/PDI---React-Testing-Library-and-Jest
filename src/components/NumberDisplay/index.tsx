import { useRef } from "react";

let idCounter = 1;

type NumberDisplayProps = {
  number: number;
};

export function NumberDisplay({ number }: NumberDisplayProps) {
  const id = useRef(idCounter++);

  return (
    <div style={{ marginBottom: "10px" }}>
      <span data-testid="number-display">{number}</span>
      <span data-testid="instance-id">{id.current}</span>
    </div>
  );
}
