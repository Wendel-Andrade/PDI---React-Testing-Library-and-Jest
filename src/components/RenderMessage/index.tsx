import { useEffect, useState } from "react";

interface RenderMessageProps {
  message: string;
}

export function RenderMessage({ message }: RenderMessageProps) {
  const [newMessage, setNewMessage] = useState<string>("");
  const [renderedMessages, setRenderedMessages] = useState<string[]>([]);

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setNewMessage(e.target.value);
  };

  useEffect(() => {
    setNewMessage(message);
  }, [message]);

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter" && newMessage.trim() !== "") {
      setRenderedMessages((prevMessages) => [...prevMessages, newMessage]);

      setNewMessage("");
    }
  };

  return (
    <div>
      <input
        type="text"
        name="message"
        placeholder="Enter the message"
        value={newMessage}
        onChange={handleInputChange}
        onKeyDown={handleKeyDown}
      />
      <p data-testid="empty-paragraph">{newMessage}</p>

      <ul style={{ padding: "0" }}>
        {renderedMessages.map((renderedMessage, index) => (
          <li key={index} style={{ listStyle: "none" }}>
            {renderedMessage}
          </li>
        ))}
      </ul>
    </div>
  );
}
