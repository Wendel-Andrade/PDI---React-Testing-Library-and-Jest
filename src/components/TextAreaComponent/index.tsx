import { ChangeEvent, useState } from "react";

interface TextAreaComponentProps {
  onSubmit: (value: string) => void;
}

export function TextAreaComponent({ onSubmit }: TextAreaComponentProps) {
  const [text, setText] = useState<string>("");

  const handleInputChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
    setText(e.target.value);
  };

  const handleSubmit = () => {
    if (text.trim() !== "") {
      onSubmit(text);
      setText("");
    }
  };

  return (
    <div
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
        gap: "8px",
      }}
    >
      <textarea
        data-testid="text-area"
        placeholder="Type your message..."
        value={text}
        onChange={handleInputChange}
      />
      <button data-testid="submit-button" onClick={handleSubmit}>
        Submit
      </button>
    </div>
  );
}
