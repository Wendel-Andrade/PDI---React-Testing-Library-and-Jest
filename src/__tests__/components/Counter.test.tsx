import { fireEvent, render, screen } from "@testing-library/react";
import { Counter } from "../../components/Counter";

describe("<Counter />", () => {
  test("renders with initial count", () => {
    render(<Counter />);

    const counterDisplay = screen.getByTestId("counter-display");

    expect(counterDisplay).toHaveTextContent("Count: 0");
  });

  test("increments count when button is clicked", () => {
    render(<Counter />);
    const counterDisplay = screen.getByTestId("counter-display");
    const incrementButton = screen.getByTestId("increment-button");

    fireEvent.click(incrementButton);

    expect(counterDisplay).toHaveTextContent("Count: 1");
  });

  test("decrement count when button is clicked", () => {
    render(<Counter />);
    const counterDisplay = screen.getByTestId("counter-display");
    const decrementButton = screen.getByTestId("decrement-button");

    fireEvent.click(decrementButton);

    expect(counterDisplay).toHaveTextContent("Count: -1");
  });

  test("reset counter count when button is clicked", () => {
    render(<Counter />);
    const counterDisplay = screen.getByTestId("counter-display");
    const resetCounterButton = screen.getByTestId("reset-counter-button");

    fireEvent.click(resetCounterButton);

    expect(counterDisplay).toHaveTextContent("Count: 0");
  });

  test("unmounts the component", () => {
    const { container, unmount } = render(<Counter />);
    const counterDisplay = screen.getByTestId("counter-display");
    const incrementButton = screen.getByTestId("increment-button");

    fireEvent.click(incrementButton);
    expect(counterDisplay).toHaveTextContent("Count: 1");

    unmount();

    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toBeNull();
  });

  test("increment snapshot", () => {
    const { asFragment } = render(<Counter />);

    const incrementButton = screen.getByTestId("increment-button");

    fireEvent.click(incrementButton);

    const componentIsFragment = asFragment();

    expect(componentIsFragment).toMatchSnapshot();
  });

  test("decrement snapshot", () => {
    const { asFragment } = render(<Counter />);

    const decrementButton = screen.getByTestId("decrement-button");

    fireEvent.click(decrementButton);

    const componentIsFragment = asFragment();

    expect(componentIsFragment).toMatchSnapshot();
  });

  test("reset counter snapshot", () => {
    const { asFragment } = render(<Counter />);

    const resetCounterButton = screen.getByTestId("reset-counter-button");

    fireEvent.click(resetCounterButton);

    const componentIsFragment = asFragment();

    expect(componentIsFragment).toMatchSnapshot();
  });
});
