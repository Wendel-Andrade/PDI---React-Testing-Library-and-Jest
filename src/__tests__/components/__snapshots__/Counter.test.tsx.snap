// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> decrement snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="increment-button"
  >
    Increment
  </button>
  <button
    data-testid="decrement-button"
  >
    Decrement
  </button>
  <button
    data-testid="reset-counter-button"
  >
    Reset Counter
  </button>
  <p
    data-testid="counter-display"
  >
    Count: -1
  </p>
</DocumentFragment>
`;

exports[`<Counter /> increment snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="increment-button"
  >
    Increment
  </button>
  <button
    data-testid="decrement-button"
  >
    Decrement
  </button>
  <button
    data-testid="reset-counter-button"
  >
    Reset Counter
  </button>
  <p
    data-testid="counter-display"
  >
    Count: 1
  </p>
</DocumentFragment>
`;

exports[`<Counter /> reset counter snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="increment-button"
  >
    Increment
  </button>
  <button
    data-testid="decrement-button"
  >
    Decrement
  </button>
  <button
    data-testid="reset-counter-button"
  >
    Reset Counter
  </button>
  <p
    data-testid="counter-display"
  >
    Count: 0
  </p>
</DocumentFragment>
`;
